CONTRIBUTING.md

```diff
@@ -78,6 +79,66 @@ use only the standard libraries, or the third-party packages we have decided to
 
 For rationale, check out the post [The Case Against Third Party Libraries](http://blog.gopheracademy.com/advent-2014/case-against-3pl/).
 
+### Writing unit tests
+Unit tests for Go code should be in the same package as the code itself.
+It should be in a file named `filename_test.go` with the package name `mypackage_test` as the name of the package.
+The tests should test functionality through public functions and methods.
+
+If access to internal members or functions is needed, then the test file should be named `filename_internal_test.go` and use the same package name as the package itself.
+The tests written in this file will have access to the internals of the package.
+This file can also be used to create testing-specific functions that can be exposed to the tests in `mypackage_test`.
+
+Pull requests require tests to be written for them and must pass our code coverage checks.
+
+### Public functions and methods
+When writing code, try to avoid making functions and methods used by the code public.
+When creating a library that needs to be used by multiple packages within the Flux repository, an internal package is preferred.
+A public function or method should only be used if using the Flux library would require the function or method to be public.
+
+### Comments and documentation
+Every new public function or method requires a valid GoDoc.
+Comments within the code should be written with proper capitalization and punctuation at the end of the sentence.
+
+### Updating markdown files for documentation
+The Flux team uses markdown for our documentation.
+When writing markdown, please use the following guidelines.
+
+- Each sentence in a paragraph is on its own line.
+- Use the `#` symbol for header sections.
+- Use the `-` symbol for bullet points.
+
+## Submitting a pull request
+To submit a pull request you should fork the Flux repository and make your change on a feature branch of your fork.
+Then generate a pull request from your branch against **master** of the Flux repository.
+Include in your pull request details of your change -- the **why** *and* the **how** -- as well as the testing you performed.
```

Comment:
@jsternberg, you write really good PR details. Maybe link here to a recent PR that you opened with a detailed comment for reference.

Reply:
It's my opinion that we shouldn't refer to a specific PR for this. I think it might be good to give some kind of guidelines, but I can't think of anything that would work well in a readme.

Reply:
Guidelines are a good idea. We can keep it how it is for now and then add these later when we see what contributors come up with.